storageservice/modules/files/file_client.bal

```diff
@@ -376,71 +376,82 @@ public isolated client class FileClient {
                 azureConfig: self.azureConfig,
                 httpVerb: http:HTTP_GET,
                 resourcePath: resourcePathForSharedkeyAuth,
                 requiredURIParameters: requiredURIParameters
             };
-            check prepareAuthorizationHeaders(authorizationDetail);      
+            check prepareAuthorizationHeaders(authorizationDetail);
         } else {
-            requestPath = requestPath.concat(QUESTION_MARK, self.azureConfig.accessKeyOrSAS.substring(1)); 
+            requestPath = requestPath.concat(QUESTION_MARK, self.azureConfig.accessKeyOrSAS.substring(1));
         }
         map<string> headerMap = populateHeaderMapFromRequest(request);
         http:Response response = check self.httpClient->get(requestPath, headerMap);
         byte[] responseBody = check response.getBinaryPayload();
         check writeFile(localFilePath, responseBody);
     }
 
-
     # Downloads a file from fileshare to a specified location as an Byte array
     #
-    # + fileShareName - Name of the FileShare
-    # + fileName - Name of the file
-    # + azureDirectoryPath - Path of azure directory
-    # + return -  If successful, file content as a byte array. Else an error
+    # + fileShareName - Name of the FileShare  
+    # + fileName - Name of the file  
+    # + azureDirectoryPath - Path of azure directory  
+    # + range - The specified byte range to be returned the file data 
+    # + return - If successful, file content as a byte array. Else an error
     @display {label: "Download File"}
-    remote isolated function getFileAsByteArray(@display {label: "File Share Name"} string fileShareName, 
-                                     @display {label: "Azure File Name"} string fileName,  
-                                     @display {label: "Azure Directory Path"} string? azureDirectoryPath = ()) 
-                                     returns @display {label: "Response"} byte[]|error {
-        string requestPath = azureDirectoryPath is () ? (SLASH + fileShareName + SLASH + fileName) : (SLASH 
-            + fileShareName + SLASH + azureDirectoryPath + SLASH + fileName);    
+    remote isolated function getFileAsByteArray(@display {label: "File Share Name"} string fileShareName,
+                                    @display {label: "Azure File Name"} string fileName,
+                                    @display {label: "Azure Directory Path"} string? azureDirectoryPath = (),
+                                    @display {label: "Specified Byte Range"} ContentRange? range = ())
+                                    returns @display {label: "Response"} byte[]|error {
+        string requestPath = azureDirectoryPath is () ? (SLASH + fileShareName + SLASH + fileName) : (SLASH
+            + fileShareName + SLASH + azureDirectoryPath + SLASH + fileName);
         http:Request request = new;
+        if (range != ()) {
+            string rangeHeader = "bytes=" + range.startByte.toString() + "-" + range.endByte.toString();
```

Comment:
Why we are getting `startByte` and `endByte` as int if we already know it needs to be converted to string?

Reply:
if `startByte` and `endByte` are defined as strings, user can pass any other invalid inputs.
Keeping it as an `int` will give a good experience comparing to `string`